packages/admin/cms-admin/src/form/file/FinalFormFileUpload.tsx

```diff
@@ -144,19 +123,7 @@ export const FinalFormFileUpload = <MaxFiles extends number | undefined>({ input
                 }
             }}
             onDownload={async (file) => {
-                setDownloadingFileIds([...downloadingFileIds, file.id]);
-                fetch(`${apiUrl}/public-upload/files/download/${file.id}`).then(async (response) => {
-                    if (!response.ok) {
-                        setDownloadingFileIds(downloadingFileIds.filter((id) => id !== file.id));
-                        setFailedToDownloadFileIds([...failedToDownloadFileIds, file.id]);
-                        return;
-                    }
-
-                    response.blob().then((blob) => {
-                        saveAs(blob, file.name);
-                        setDownloadingFileIds(downloadingFileIds.filter((id) => id !== file.id));
-                    });
-                });
+                window.open(`${apiUrl}/public-upload/files/download/${file.id}`);
```

Comment:
It would be nicer if this were a real `<a>` tag, then the browser would download it directly, and the user wouldn't have to switch tabs and download it there. This could become tedious, especially if there are many files to downloads.

Of course, this doesn't work nicely with your generic component...

Reply:
Good point!

> Of course, this doesn't work nicely with your generic component...

If there is a use-case for the current `onDownload` we can add `downloadUrl` as an additional, alternative prop. 
But maybe we don't even need `onDownload` anyway 🤔

Reply:
I added a `downloadUrl` prop to `FileSelectListItem` and a `getDownloadUrl` prop to `FileSelect` to allow a callback to access the `file` value when creating the url.